Use severe notification for project destroy dialog

oliverguenther opened the pull request:
The current project `destroy_info` style is inconsistent with the
styleguide, as it uses a `notification` (used for one-line notifications
only) instead of a warning notification box.

This commit switches this dialog to the notification-box with warning
parameter and suggests to additionaly use the new `-severe` switch for warnings
due the potential ramifications this action has.

Using `-severe` also makes it compatible with the upcoming repository
destroy_info dialog, which also uses this dialog when local data is
to be deleted outside OpenProject.

Visual appearance:

![bildschirmfoto 2015-07-23 um 11 12 49](https://cloud.githubusercontent.com/assets/459462/8847076/d45bab36-312b-11e5-8a1e-9d76497187b7.png)

oliverguenther committed:
Use severe notification for project destroy dialog

The current project `destroy_info` style is inconsistent with the
styleguide, as it uses a `notification` (used for one-line notifications
only) instead of a warning notification box.

This commit switches this dialog to the notification-box with warning
parameter and suggests to additionaly use the new `-severe` switch for warnings
due the potential ramifications this action has.

Using `-severe` also makes it compatible with the upcoming repository
destroy_info dialog, which also uses this dialog when local data is
to be deleted outside OpenProject.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-feature-project-destroy-notification.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://oliverguenther-feature-project-destroy-notification.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

floriank commented on app/views/projects/destroy_info.html.erb:
```diff
@@ -27,18 +27,20 @@ See doc/COPYRIGHT.rdoc for more details.
+      <% end %>
+    </p>
+    <p>
+      <%= form_tag(project_path(@project_to_destroy), :method => :delete) do %>
```
is there a reason not to use `styled_form_tag`? :wink:

floriank replied:
Also: awesome JSONesque syntax for Hashes is awesome.

floriank commented on app/views/projects/destroy_info.html.erb:
```diff
@@ -27,18 +27,20 @@ See doc/COPYRIGHT.rdoc for more details.
+    <p>
+      <%= form_tag(project_path(@project_to_destroy), :method => :delete) do %>
+        <label><%= check_box_tag 'confirm', 1 %> <%= l(:general_text_Yes) %></label>
+        <%= submit_tag l(:button_delete), class: 'button -highlight' %>
```
this might clash with the red of the severe warning.

NobodysNightmare replied:
This is where we will always struggle with accessibility vs. themability...

You mean that the button background-color might clash with the background-color of the notification in case anyone (no names ^^) has a red-ish highlight button, don't you?

Not sure how we can solve this problem in general...

oliverguenther replied:
In this case we can work around it by removing `-highlight`, as it is the only button in the field (and this is consistent with the styleguide error notification-box example, which also features a default button).

NobodysNightmare commented on config/locales/en.yml:
```diff
@@ -1392,7 +1392,8 @@ en:
   text_powered_by: "Powered by %{link}"
-  text_project_destroy_confirmation: "Are you sure you want to delete this project and related data?"
+  text_project_destroy_title: "Do you really want to delete this project?"
+  text_project_destroy_confirmation: "If you continue, the project %{identifier} and all related data will be perminently destroyed. Please check the box below if you want to continue."
```
_permanently_

![](http://i.imgur.com/QTgi1pH.jpg)

oliverguenther committed:
Correct for review

- Switch to styled_form and remove highlight from button
- Fix permanently typo

myabc commented:
@floriank the blue delete button doesn't look good. try making it red?

oliverguenther commented:
@myabc The styleguide suggests to use the default button, I've updated this PR to reflect it (though the image above is outdated, I'll change it.)

oliverguenther commented:
![bildschirmfoto 2015-07-23 um 11 12 49](https://cloud.githubusercontent.com/assets/459462/8847076/d45bab36-312b-11e5-8a1e-9d76497187b7.png)

floriank commented:
@myabc yes i noted that on the line already - I think the styleguide dictates a default button.

One style note from me - could we place the check box next to the button instead of above?

ulferts commented:
While I also think there is room for improvements, I think we shouldn't put too much emphasis on this one here. It's way better than before.
